pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -144,35 +147,178 @@ func (c *Controller) enqueueNode(obj interface{}) {
+// the cluster. If the antrea agent restarts and Nodes have left the cluster, this function will
+// take care of removing routes which are no longer valid.
+func (c *Controller) removeStaleGatewayRoutes() error {
+	nodes, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error when listing Nodes: %v", err)
+	}
+
+	routes, err := c.listRoutes()
+	if err != nil {
+		return fmt.Errorf("error when listing existing gateway routes: %v", err)
+	}
+
+	for _, node := range nodes {
+		_, podCIDR, err := net.ParseCIDR(node.Spec.PodCIDR)
+		if err != nil {
+			return fmt.Errorf("error when parsing Pod CIDR for Node %s: %v", node.Name, err)
+		}
+		podCIDRStr := podCIDR.String()
+		// remove from map of routes: at the end of the loop the remaining routes will be
+		// the stale ones (the ones we want to delete).
+		// we iterate over all current Nodes, including the Node on which this agent is
+		// running, so the route to local Pods will be removed from the map as well.
+		delete(routes, podCIDRStr)
+	}
+
+	// remove all remaining routes in the map: these routes do not match any Node in the
+	// cluster.
+	klog.V(4).Infof("Need to delete %d routes", len(routes))
+	for _, route := range routes {
+		if err := netlink.RouteDel(route); err != nil {
+			klog.Errorf("error when deleting route '%v': %v", route, err)
+		}
+	}
+
+	return nil
+}
+
+// removeStaleTunnelPorts removes all the tunnel ports which no longer correspond to a Node in the
+// cluster. If the antrea agent restarts and Nodes have left the cluster, this function will take
+// care of removing tunnel ports which are no longer valid. If the tunnel port configuration has
+// changed, the tunnel port will also be deleted (the controller loop will later take care of
+// re-creating the port with the correct configuration).
+func (c *Controller) removeStaleTunnelPorts() error {
+	nodes, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error when listing Nodes: %v", err)
+	}
+	// desiredInterfaces is the set of interfaces we wish to have, based on the current list of
+	// Nodes. If a tunnel port corresponds to a valid Node but its configuration is wrong, we
+	// will not include it in the set.
+	desiredInterfaces := make(map[string]bool)
+	// knownInterfaces is the list of interfaces currently in the local cache.
+	knownInterfaces := c.interfaceStore.GetInterfaceKeysByType(interfacestore.TunnelInterface)
+
+	if c.ipsecPSK != "" {
+		for _, node := range nodes {
+			interfaceConfig, found := c.interfaceStore.GetNodeTunnelInterface(node.Name)
+			if !found {
+				// Tunnel port not created for this Node, nothing to do.
+				continue
+			}
+
+			peerNodeIP, err := getNodeAddr(node)
+			if err != nil {
+				klog.Errorf("Failed to retrieve IP address of Node %s: %v", node.Name, err)
+				continue
+			}
+
+			portName := util.GenerateNodeTunnelInterfaceName(node.Name)
```

Comment:
I think you should use GenerateNodeTunnelInterfaceKey() here, as InterfaceName no more matches the interface store map key now.

Reply:
yes, you're correct, that's a typo